config/tests/samples/create/harness.go

```diff
@@ -97,33 +98,38 @@ type Harness struct {
 }
 
 type HarnessOptions struct {
-	VCRPath string
+	VCRPath    string
+	RunSamples bool
 }
 
 type httpRoundTripperKeyType int
 
 // httpRoundTripperKey is the key value for http.RoundTripper in a context.Context
 var httpRoundTripperKey httpRoundTripperKeyType
 
-// NewHarnessWithManager builds a Harness for an existing manager.
-// deprecated: Prefer NewHarness, which can construct a manager and mock gcp etc.
-func NewHarnessWithManager(ctx context.Context, t *testing.T, mgr manager.Manager) *Harness {
-	h := &Harness{
-		T:      t,
-		Ctx:    ctx,
-		client: mgr.GetClient(),
+func NewHarness(ctx context.Context, t *testing.T) *Harness {
+	return newHarnessWithOptions(ctx, t, &HarnessOptions{})
+}
+
+func NewVCRHarness(ctx context.Context, t *testing.T, path string) *Harness {
+	opts := &HarnessOptions{
+		VCRPath: path,
 	}
-	return h
+	return newHarnessWithOptions(ctx, t, opts)
 }
 
-func NewHarness(ctx context.Context, t *testing.T) *Harness {
+func NewSamplesHarness(ctx context.Context, t *testing.T) *Harness {
 	opts := &HarnessOptions{
-		VCRPath: "",
+		RunSamples: true,
+	}
+	err := os.Setenv("E2E_KUBE_TARGET", "envtest")
+	if err != nil {
+		t.Errorf("error setting kube target: %v", err)
 	}
-	return NewHarnessWithOptions(ctx, t, opts)
+	return newHarnessWithOptions(ctx, t, opts)
 }
 
-func NewHarnessWithOptions(ctx context.Context, t *testing.T, opts *HarnessOptions) *Harness {
+func newHarnessWithOptions(ctx context.Context, t *testing.T, opts *HarnessOptions) *Harness {
```

Comment:
It's probably out of scope, but this function is super long, and we should try to refactor / shorten it whenever we have a chance.